Tests/SKCoreTests/TaskSchedulerTests.swift

```diff
@@ -0,0 +1,411 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SKCore
+import XCTest
+
+final class TaskSchedulerTests: XCTestCase {
+  func testHighPriorityTasksGetExecutedBeforeLowPriorityTasks() async throws {
+    let highPriorityTasks: Int = 4
+    let lowPriorityTasks: Int = 2
+    await runTaskScheduler(
+      highPriorityTasks: highPriorityTasks,
+      lowPriorityTasks: lowPriorityTasks,
+      scheduleTasks: { scheduler, taskExecutionRecorder in
+        for i in 0..<20 {
+          let id = TaskID.lowPriority(i)
+          await scheduler.schedule(priority: .low, id: id) {
+            await taskExecutionRecorder.run(taskID: id)
+          }
+        }
+
+        for i in 0..<10 {
+          let id = TaskID.highPriority(i)
+          await scheduler.schedule(priority: .high, id: id) {
+            await taskExecutionRecorder.run(taskID: id)
+          }
+        }
+      },
+      validate: { (recordings: [Set<TaskID>]) in
+        // Check that all high-priority tasks get executed before the low-priority tasks
```

Comment:
We *could* be executing some high priority and low priority at the same time in the actual check (eg. 2 high -> high + low -> low + low would/should pass).